docs/reduction-in-locks.md

```diff
@@ -0,0 +1,107 @@
+
+* The redo log tracks all file operations to maintain consistency. The backup process generates new metadata files to handle any DDL operations that occur during the backup. When you run `--prepare`, the `xtrabackup` first processes these metadata files to ensure a consistent database state with respect to file-level operations. Then the redo and undo logs are used to make the backup fully consistent.
+
+### Limitations
+
+* Certain DDL operations consume significant resources, and xtrabackup simultaneously requires I/O to copy and read files. This requirement can increase resource demand from DDL operations during the backup process.
+
+* Backups may be larger because new `DDL` operations are executed concurrently on the server, and the files generated by the server are included in the backup. Additionally, certain `DDL` operations, such as `ADD INDEX` or encryption changes to existing data files, will cause the data files to be recopied, increasing the backup size.
+
+* The number of open file handles in your operating system should be configured to match the number of files in the server data directory.
+
+## How the backup lock with the `--lock-ddl` option works
+
+Percona Server for MySQL implemented `LOCK TABLES FOR BACKUP` to block `DDL` operations on a server. Percona `xtrabackup` utilizes this lock during the backup process to ensure that any `DDL` operation does not corrupt the backup. With MySQL-compatible servers `xtrabackup` utilizes `Lock Instance For Backup` to lock the server during the backup. The lock does not affect the `Data Manipulation Language` (DML) operations.
+
+With the [`--lock-ddl=ON`](./xtrabackup-option-reference.md/#lock-ddl) option the backup lock is enabled by default from the start of the backup process.
+{ .power-number }
+
+1. Set a backup lock on the server to prevent any DDL operations, such as creating or altering tables.
+
+2. Parse and copy the redo logs from the checkpoint up to the current `Log Sequence Number` (LSN), and start tracking new file operations.
+
+3. Identify and copy the `.ibd` files.
+
+4. Copy `non-InnoDB` files.
+
+5. Gather a sync point from all engines (InnoDB LSN, binlog, Global Transaction Identifier (GTID), etc.) by querying the log status.
+
+6. Stop the redo thread once it copies at least up to sync point.
+
+7. Release the backup lock on the server.
+
+If the backup lock is disabled with the [`--lock-ddl=OFF`](./xtrabackup-option-reference.md/#lock-ddl) option, a backup continues while concurrent DDL operations are executed. These backups may be invalid and may fail at either the `backup` or the `--prepare` step.
+
+## How the `--lock-ddl=REDUCED` option works
+
+Taking backup with the `--lock-ddl=REDUCED` option includes the following phases:
+
+### Phase 1: Operations performed without the lock
+
+!!! warning
+
+    The `ALTER INSTANCE ROTATE MASTER KEY` command should not be executed during Phase 1 if there are encrypted tables, as it will cause the backup to fail.
```

Comment:
Should be 
Do not execute the ALTER INSTANCE ROTATE MASTER KEY command during Phase 1 if there are encrypted tables. This will prevent backup failures.

Reply:
done